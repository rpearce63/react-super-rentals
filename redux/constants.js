export const LOAD_RENTALS = "LOAD_RENTALS";
export const FILTER_VALUE = "FILTER_VALUE";
export const FIND_BY_ID = "FIND_BY_ID";
