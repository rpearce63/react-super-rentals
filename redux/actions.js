/*
 * You can put your actions here in this file, or break them up across
 * multiple files. Redux patterns are very... flexible. This is just one
 * possible approach.
 *
 * You should obviously replace this action with some of your own that actually
 * do useful things.
 *
 * For more information please see the Redux Actions Documentation
 * https://redux.js.org/basics/actions
 */

import * as constants from "./constants";
import { rentalData } from "./rentalData";

export const loadRentals = () => {
  return {
    type: constants.LOAD_RENTALS,
    payload: rentalData
  };
};

export const setListFilter = param => {
  return {
    type: constants.FILTER_VALUE,
    payload: param
  };
};
