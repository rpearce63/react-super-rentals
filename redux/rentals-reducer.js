import * as constants from "./constants";

export default function(state = { rentalData: [], filter: "" }, action) {
  switch (action.type) {
    case constants.LOAD_RENTALS:
      return { ...state, rentalData: action.payload };
    case constants.FILTER_VALUE:
      return { ...state, filter: action.payload };
    default:
      return state;
  }
}
