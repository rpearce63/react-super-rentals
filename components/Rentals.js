import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { NavLink } from "react-router-dom";
import { loadRentals } from "../redux/actions";
import RentalListing from "./rental-listing";
import { ListFilter } from "./components";
import { filterList } from "../redux/rentalData";

export class Rentals extends Component {
  componentDidMount() {
    this.props.loadRentals();
  }

  render() {
    const filteredList = filterList(
      this.props.listFilter,
      this.props.rentalData
    );
    return (
      <div>
        <div className="jumbo">
          <div className="right tomster" />
          <h2>Welcome!</h2>
          <p>
            We hope you find exactly what you're looking for in a place to stay.
          </p>
          <NavLink to="/about" className="button">
            About Us
          </NavLink>
        </div>
        <br />
        <ListFilter>
          {filteredList.length > 0 ? (
            filteredList.map((rental, index) => (
              <RentalListing key={index} rental={rental} />
            ))
          ) : (
            <div className="no-matches">
              <h3>No matches found</h3>
            </div>
          )}
        </ListFilter>
      </div>
    );
  }
}
function mapDispatchToProps(dispatch) {
  return bindActionCreators({ loadRentals }, dispatch);
}

function mapStateToProps(state) {
  return {
    rentalData: state.rentals.rentalData,
    listFilter: state.rentals.filter
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Rentals);
