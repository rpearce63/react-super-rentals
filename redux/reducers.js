import { combineReducers } from "redux";

import RentalsReducer from "./rentals-reducer";

const rootReducer = combineReducers({
  rentals: RentalsReducer
});

export default rootReducer;
