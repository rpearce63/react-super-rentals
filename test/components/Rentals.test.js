import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Rentals from "../../components/Rentals";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { rentalData } from "../../redux/rentalData";

const initialState = { rentals: rentalData };

const mockStore = configureStore();
let wrapper;
let store;

beforeEach(() => {
  store = mockStore(initialState);
  wrapper = shallow(
    <Provider store={store}>
      <Rentals />
    </Provider>
  );
});

Enzyme.configure({ adapter: new Adapter() });

describe("Rental list tests", () => {
  it("renders the Rentals page", () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
