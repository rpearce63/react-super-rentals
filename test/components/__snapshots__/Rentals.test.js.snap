// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rental list tests renders the Rentals page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(Rentals) />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Connect(Rentals) />,
      "value": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeState": Object {
          "rentals": Array [
            Object {
              "bedrooms": 15,
              "category": "Estate",
              "city": "San Francisco",
              "description": "This grand old mansion sits on over 100 acres of rolling hills and dense redwood forests.",
              "id": "grand-old-mansion",
              "image": "https://upload.wikimedia.org/wikipedia/commons/c/cb/Crane_estate_(5).jpg",
              "owner": "Veruca Salt",
              "title": "Grand Old Mansion",
            },
            Object {
              "bedrooms": 1,
              "category": "Condo",
              "city": "Seattle",
              "description": "A commuters dream. This rental is within walking distance of 2 bus stops and the Metro.",
              "id": "urban-living",
              "image": "https://upload.wikimedia.org/wikipedia/commons/0/0e/Alfonso_13_Highrise_Tegucigalpa.jpg",
              "owner": "Mike TV",
              "title": "Urban Living",
            },
            Object {
              "bedrooms": 3,
              "category": "Apartment",
              "city": "Portland",
              "description": "Convenience is at your doorstep with this charming downtown rental. Great restaurants and active night life are within a few feet.",
              "id": "downtown-charm",
              "image": "https://upload.wikimedia.org/wikipedia/commons/f/f7/Wheeldon_Apartment_Building_-_Portland_Oregon.jpg",
              "owner": "Violet Beauregarde",
              "title": "Downtown Charm",
            },
          ],
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Connect(Rentals) />,
        "value": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "storeState": Object {
            "rentals": Array [
              Object {
                "bedrooms": 15,
                "category": "Estate",
                "city": "San Francisco",
                "description": "This grand old mansion sits on over 100 acres of rolling hills and dense redwood forests.",
                "id": "grand-old-mansion",
                "image": "https://upload.wikimedia.org/wikipedia/commons/c/cb/Crane_estate_(5).jpg",
                "owner": "Veruca Salt",
                "title": "Grand Old Mansion",
              },
              Object {
                "bedrooms": 1,
                "category": "Condo",
                "city": "Seattle",
                "description": "A commuters dream. This rental is within walking distance of 2 bus stops and the Metro.",
                "id": "urban-living",
                "image": "https://upload.wikimedia.org/wikipedia/commons/0/0e/Alfonso_13_Highrise_Tegucigalpa.jpg",
                "owner": "Mike TV",
                "title": "Urban Living",
              },
              Object {
                "bedrooms": 3,
                "category": "Apartment",
                "city": "Portland",
                "description": "Convenience is at your doorstep with this charming downtown rental. Great restaurants and active night life are within a few feet.",
                "id": "downtown-charm",
                "image": "https://upload.wikimedia.org/wikipedia/commons/f/f7/Wheeldon_Apartment_Building_-_Portland_Oregon.jpg",
                "owner": "Violet Beauregarde",
                "title": "Downtown Charm",
              },
            ],
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
